pkg/proctree/exit_processing.go

```diff
@@ -0,0 +1,96 @@
+package proctree
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/tracee/signatures/helpers"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// processExitEvent remove references of processes from the tree when the corresponding process exit without children, or
+// if the last child process of a process exits.
+// Notice that there is a danger of memory leak if there are lost events of sched_process_exit (but is limited to the
+// possible number of PIDs - 32768)
+func (tree *ProcessTree) processExitEvent(event *trace.Event) error {
+	err := tree.processDefaultEvent(event)
```

Comment:
Just to be sure, we're using this as an opportunity to fill out the process tree from the terminating process which generated the event?